src/core/Rational.pm6

```diff
@@ -94,35 +94,38 @@ my role Rational[::NuT = Int, ::DeT = ::("NuT")] does Real {
         my $whole  = self.abs.floor;
         my $fract  = self.abs - $whole;
 
+        my $reduce = False;
         # fight floating point noise issues RT#126016
-        if $fract.Num == 1e0 { ++$whole; $fract = 0 }
+        if $fract.Num == 1e0 {
+            if nqp::unbox_s(self.^name) eq 'Rat' {
+                $whole += 1;
+                $fract = 0
```

Comment:
This is kind of a nitpick, but I think there's a convention that last statements in blocks only leave off the semi-colon if the value is being returned.

Reply:
Fair enough.